packages/core/src/actors.ts

```diff
@@ -344,102 +381,30 @@ export function fromEventObservable<T extends EventObject>(
         default:
           return state;
       }
     },
-    initialState: undefined
+    initialState: {
+      subscription: undefined as Subscription | undefined,
+      observable: undefined as Subscribable<T> | undefined,
+      canceled: false,
+      data: undefined
+    },
+    getSnapshot: (_) => undefined
   };
 
   return behavior;
 }
 
 export function fromMachine<TMachine extends AnyStateMachine>(
-  machine: AreAllImplementationsAssumedToBeProvided<
-    TMachine['__TResolvedTypesMeta']
-  > extends true
-    ? TMachine
-    : 'Some implementations missing',
-  options: Partial<InterpreterOptions> = {}
-): Behavior<EventFrom<TMachine>, StateFrom<TMachine>> {
-  const snapshotEventType = Symbol('snapshot');
-
-  const castedMachine = machine as TMachine;
-  let service: InterpreterFrom<TMachine> | undefined;
-  let initialState: StateFrom<TMachine>;
-
-  // TODO: use better type for `TEvent`
-  const behavior: Behavior<any, StateFrom<TMachine>> = {
-    transition: (state, event, actorContext) => {
-      const { _parent: parent } = actorContext.self;
-
-      switch (event.type) {
-        case startSignalType:
-          service = interpret(castedMachine as AnyStateMachine, {
-            ...options,
-            parent,
-            id: actorContext.name
-          }) as InterpreterFrom<TMachine>;
-          service.onDone((doneEvent) => {
-            parent?.send(
-              toSCXMLEvent(doneEvent, {
-                origin: actorContext.self
-              })
-            );
-          });
-
-          service.subscribe((state) => {
-            actorContext.self.send({
-              type: snapshotEventType,
-              snapshot: state
-            });
-          });
-          service.start();
-          return state;
-        case stopSignalType:
-          service?.stop();
-          return state;
-        case snapshotEventType: {
-          const snapshot = event.snapshot;
-          if (options.sync) {
-            parent?.send(
-              toSCXMLEvent(
-                {
-                  type: actionTypes.update,
-                  state: snapshot
-                },
-                { origin: actorContext.self }
-              )
-            );
-          }
-          return snapshot;
-        }
-        default:
-          const _event = actorContext._event;
-
-          if (isSignal(_event)) {
-            // TODO: unrecognized signal
-            return state;
-          }
-
-          service?.send(_event);
-          return state;
-      }
-    },
-    get initialState() {
-      // TODO: recheck if this caching is needed, write a test for its importance or remove the caching
-      if (initialState) {
-        return initialState;
-      }
-      initialState = castedMachine.getInitialState() as StateFrom<TMachine>;
-      return initialState;
-    }
-  };
-
-  return behavior;
+  machine: TMachine,
+  _options: Partial<InterpreterOptions> = {}
+): TMachine {
+  return machine;
```

Comment:
Pure simplicity; a machine is already a behavior 😌 

Maybe we don't need `fromMachine`, I'll look into removing it.